ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModel.kt

```diff
@@ -69,51 +67,42 @@ constructor(
     ioDispatcher
   ) {
 
-  private val fragmentArgs: @Hot(replays = true) PublishSubject<OfflineAreaViewerFragmentArgs> =
-    PublishSubject.create()
-  private val removeAreaClicks: @Hot PublishSubject<Nil> = PublishSubject.create()
-
-  private val context: WeakReference<Context>
-
   /** Returns the offline area associated with this view model. */
-  @JvmField val offlineArea: LiveData<OfflineArea>
-  @JvmField var areaName: LiveData<String>
+  val area = MutableLiveData<OfflineArea>()
+  val areaName = area.map { it.name }
+  val areaSize = MutableLiveData<String>()
+  val progressOverlayVisible = MutableLiveData<Boolean>()
 
   private var offlineAreaId: String? = null
 
   override val mapConfig: MapConfig
     get() =
       super.mapConfig.copy(
-        showOfflineTileOverlays = false,
-        overrideMapType = MapType.ROAD,
+        showOfflineTileOverlays = true,
+        overrideMapType = MapType.TERRAIN,
         disableGestures = true
       )
 
-  init {
-    this.context = WeakReference(context)
-
-    // We only need to convert this single to a flowable in order to use it with LiveData.
-    // It still only contains a single offline area returned by getOfflineArea.
-    val offlineAreaItemAsFlowable: @Hot Flowable<OfflineArea> =
-      this.fragmentArgs
-        .map(OfflineAreaViewerFragmentArgs::getOfflineAreaId)
-        .flatMapSingle(offlineAreaRepository::getOfflineArea)
-        .doOnError { Timber.e(it, "Couldn't render area %s", offlineAreaId) }
-        .toFlowable(BackpressureStrategy.LATEST)
-
-    areaName = offlineAreaItemAsFlowable.map(OfflineArea::name).toLiveData()
-    offlineArea = offlineAreaItemAsFlowable.toLiveData()
+  /** Initialize the view model with the given arguments. */
+  fun initialize(args: OfflineAreaViewerFragmentArgs) {
+    offlineAreaId = args.offlineAreaId
```

Comment:
Can we make `offlineAreaId` non-nullable since we are asserting it non-null on line 90 anyways?